crates/config/src/sections/upstream_oauth2.rs

```diff
@@ -565,4 +565,11 @@ pub struct Provider {
     /// Orders of the keys are not preserved.
     #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
     pub additional_authorization_parameters: BTreeMap<String, String>,
+
+    /// Whether the login_hint should be forwarded to the provider in the
+    /// authorization request.
+    ///
+    /// Defaults to `false`.
+    #[serde(default)]
+    pub forward_login_hint: bool,
```

Comment:
I imagine this should be a template to allow people to configure what exactly gets forwarded based on their upstream provider's requirements

Reply:
I think later on we'd want a behaviour which looks up the user based on a `mxid:` login hint, and forwards to the upstream provider with either a `id_token_hint` (the last id_token we've seen for this user) or a templated `login_hint`. This obviously has some privacy implications, which is why we want to be careful with this

But for now, this simple feature of just forwarding the login_hint as-is could already be valuable for some deployments